test/org/sagebionetworks/bridge/cache/ViewCacheTest.java

```diff
@@ -0,0 +1,144 @@
+                @Override public Study get() {
+                    throw new BridgeServiceException("There has been a problem retrieving the study");
+                }
+            });
+        } catch(BridgeServiceException e) {
+            assertEquals("There has been a problem retrieving the study", e.getMessage());
+        }
+    }
+    
+    @Test
+    public void somethingIsCached() throws Exception {
+        String cacheKey = getCacheKey(Study.class, study.getIdentifier());
+        String originalStudyJson = mapper.writeValueAsString(study);
+        
+        CacheProvider provider = mock(CacheProvider.class);
+        when(provider.getString(cacheKey)).thenReturn(originalStudyJson);
+        
+        ViewCache cache = new ViewCache();
+        cache.setCacheProvider(provider);
+        
+        String json = cache.getView(Study.class, study.getIdentifier(), new Supplier<Study>() {
+            @Override public Study get() {
+                Study study = new DynamoStudy();
+                study.setName("Test Study 2");
+                return study;
+            }
+        });
+        
+        Study foundStudy = DynamoStudy.fromJson(mapper.readTree(json));
+        assertEquals("Test Study", foundStudy.getName());
+    }
+    
+    @Test
+    public void removeFromCacheWorks() throws Exception {
+        ViewCache cache = new ViewCache();
+        
+        final String originalStudyJson = mapper.writeValueAsString(study);
+        final String cacheKey = getCacheKey(Study.class, study.getIdentifier());
+        cache.setCacheProvider(getSimpleCacheProvider(cacheKey, originalStudyJson));
+        
+        cache.removeView(Study.class, study.getIdentifier());
+        
+        String json = cache.getView(Study.class, study.getIdentifier(), new Supplier<Study>() {
+            @Override public Study get() {
+                Study study = new DynamoStudy();
+                study.setName("Test Study 2");
+                return study;
+            }
+        });
+        Study foundStudy = DynamoStudy.fromJson(mapper.readTree(json));
+        assertEquals("Test Study 2", foundStudy.getName());
+    }
+    
+    private CacheProvider getSimpleCacheProvider(final String cacheKey, final String originalStudyJson) {
+        return new CacheProvider() {
+            private Map<String,String> map = Maps.newHashMap();
+            {
+                map.put(cacheKey, originalStudyJson);
+            }
+            public String getString(String cacheKey) {
+                return map.get(cacheKey);
+            }
+            public void setString(String cacheKey, String value, int ttlSeconds) {
+                map.put(cacheKey, value);
+            }
+            public void removeString(String cacheKey) {
+                map.remove(cacheKey);
+            }
+        };   
+    }
+    
+    private String getCacheKey(Class<?> clazz, String id) {
+        return RedisKey.VIEW.getRedisKey(id + ":" + clazz.getSimpleName());
+    }
```

Comment:
Yes. I changed ViewCache so it has a method, getCacheKey() that returns a ViewCacheKey, and that is used in the other signatures. Now all code uses that method to construct the key.